lib/std/c/darwin.zig

```diff
@@ -3085,3 +3094,471 @@ pub const PT_DENY_ATTACH = 31;
 pub const caddr_t = ?[*]u8;
 
 pub extern "c" fn ptrace(request: c_int, pid: pid_t, addr: caddr_t, data: c_int) c_int;
+
```

Comment:
I am not sure I follow. Why would we want any binding for `ptrace` here and not directly in `os.zig`?

Reply:
I am happy with adding separating comments but I think it's somewhat premature - I don't even yet know how to structure this file, and currently it's a bit of a mess. The reason I created this PR was revert an erroneous decision by me of introducing `os/darwin.zig` which seems backwards as everything on macOS is done via `libSystem` anyhow. Also, this way we avoid unnecessary and pointless `usingnamespace`s.

Reply:
> Why would we want any wrapper for ptrace here and not directly in os.zig?

Since the function name is identical (+ we have no posix.zig yet to distinguish unifiable and non-unifiable posix semantics), definitely yes. If the function name is not identical or it would offer semantics not available on other OSes, then this would sound less good.

> I don't even yet know how to structure this file, and currently it's a bit of a mess

Ah, kk.